concrete-core/src/backends/core/private/crypto/ggsw/fourier_seeded.rs

```diff
@@ -0,0 +1,657 @@
+use crate::backends::core::private::crypto::bootstrap::FourierBuffers;
+use crate::backends::core::private::crypto::glwe::{GlweCiphertext, GlweSeededList};
+use crate::backends::core::private::crypto::secret::generators::EncryptionRandomGenerator;
+use crate::backends::core::private::math::fft::{Complex64, FourierPolynomial};
+use crate::backends::core::private::math::polynomial::Polynomial;
+use crate::backends::core::private::math::random::RandomGenerator;
+use crate::backends::core::private::math::tensor::{
+    ck_dim_div, AsMutSlice, AsMutTensor, AsRefSlice, AsRefTensor, IntoTensor, Tensor,
+};
+use crate::backends::core::private::math::torus::UnsignedTorus;
+
+use super::{FourierGgswCiphertext, GgswSeededLevelMatrix, StandardGgswSeededCiphertext};
+
+use crate::backends::core::private::math::decomposition::DecompositionLevel;
+use concrete_commons::parameters::{
+    DecompositionBaseLog, DecompositionLevelCount, GlweSize, PolynomialSize,
+};
+use concrete_fftw::array::AlignedVec;
+#[cfg(feature = "multithread")]
+use rayon::{iter::IndexedParallelIterator, prelude::*};
+
+/// A GGSW ciphertext in the Fourier Domain.
+#[derive(Debug, Clone, PartialEq)]
+pub struct FourierGgswSeededCiphertext<Cont, Scalar> {
+    tensor: Tensor<Cont>,
+    poly_size: PolynomialSize,
+    glwe_size: GlweSize,
+    decomp_base_log: DecompositionBaseLog,
+    _scalar: std::marker::PhantomData<Scalar>,
+    seed: u128,
+    shift: usize,
+}
+
```

Comment:
Is this type really needed ? If I understand correctly, its purpose is to basically save the transformation of a few polynomials (the bodies) which are already stored in the fourier domain. That makes a lot of code to maintain, for something which does not seem really necessary. Indeed, decompression is a resource intensive operation, but it is not expected to happen in the middle of a computation. I imagine, we will only do it once after the things have been transmitted. It seems to me that for now, it would be enough to have an entry point which decompresses the `StandarSeededGgswCiphertext` directly into a `FourierGgswCiphertext`. We could postpone bringing the fourier-seeded types, waiting to see if the decompression is really a big concern and depending on that, choose to bring those or not. What do you think ?

Reply:
Of course, the same goes for all the other fourier-seeded types.

Reply:
Decompression will not happen mid-computation, but it does not mean it will only happen once: the server is very likely willing to store the public keys under their seeded form and expand them only when using them. So I think it is useful to have a seeded format in the fourier domain.